core/math/transform_interpolator.cpp

```diff
@@ -342,10 +342,7 @@ bool TransformInterpolator::_basis_is_orthogonal(const Basis &p_basis, real_t p_
 	Basis m = p_basis * p_basis.transposed();
 
 	// Less stringent tests than the standard Godot slerp.
-	if (!_vec3_is_equal_approx(m[0], identity[0], p_epsilon) || !_vec3_is_equal_approx(m[1], identity[1], p_epsilon) || !_vec3_is_equal_approx(m[2], identity[2], p_epsilon)) {
-		return false;
-	}
-	return true;
+	return !!_vec3_is_equal_approx(m[0], identity[0], p_epsilon) || !_vec3_is_equal_approx(m[1], identity[1], p_epsilon) || !_vec3_is_equal_approx(m[2], identity[2], p_epsilon);
```

Comment:
`!!`?

Reply:
> `!!`?

the syntax ensures the value is of boolean type. !! (not not) == true

Reply:
This would have to be !(!x || !y || !z) with parentheses, which can be simplified to x && y && z

Reply:
`!!` is kinda familiar in web development. Not much in C++ and even less in the Godot codebase. In C++ we have casting (`(bool)`).